osquery/core/posix/utils.cpp

```diff
@@ -41,4 +41,33 @@ Status platformStrncpy(char* dst, size_t nelms, const char* src, size_t count) {
   ::strncpy(dst, src, count);
   return Status(0, "OK");
 }
+
+char* canonicalize_file_name(char* name) {
+  // This implementaion mimic behaviour of realpath
+  // function with NULL as buffer, except the fact
+  // that default fallback buffer is 4096 instead of 1024
+  // since PATH_MAX default value is usually 4096+
+  // In modern versions of libc pathing PATH_MAX buffer
+  // is safe and will be handled corrrectly.
+  // There is no strong evidence that using PATH_MAX can lead to
+  // buffer overflow on supported OS versions
+
+  long int path_max = 0;
+#ifdef PATH_MAX
+  path_max = PATH_MAX;
+#else
+  path_max = pathconf(name, _PC_PATH_MAX);
+  if (path_max <= 0) {
+    path_max = 1024 * 4;
+  }
+#endif
+  char* buffer = reinterpret_cast<char*>(malloc(path_max));
```

Comment:
You don't need malloc, just char buffer[path_max] works after C99 I think.

Reply:
Oh you are returning that. But anyway they always cast it to string and then free, why not just return string?